### PR TITLE
bug/1426

### PR DESCRIPTION
**Pull request by brouillette**
- added fetchingVariantsOverview to explorer state
- infer num of variants ~according to each assemblyId~ based on the number of sampleIDs

### CODE REVIEW

File: src/components/overview/VariantsSummary.js [hunk 1]
```diff
@@ -4,17 +4,24 @@ import { Col, Row, Spin, Statistic, Typography, Icon } from "antd";
 
 const VariantsSummary = ( ) => {
 
-    const fetchingTableSummaries = useSelector(state => state.tableSummaries?.isFetching);
-    const numVCFs =
-        useSelector((state) => state.overviewSummary.data?.data_type_specific?.experiment_results.file_format.VCF) || 0;
+    const fetchingVariantsOverview = useSelector(state => state.explorer?.fetchingVariantsOverview);
+    const variantsOverviewResults = useSelector((state) => state.explorer.variantsOverviewResponse);
```
Comment (gsfk) [inline src/components/overview/VariantsSummary.js:8]: not sure here why one line has optional chaining and the other doesn't...

**Review by gsfk — COMMENTED** [review]
Looks good, although gohan won't let me test right now. Also, when gohan fails, the spinner runs forever, not sure if this is a bug or a feature.

File: src/components/overview/VariantsSummary.js (hunk 1 above)
Comment (davidlougheed) [reply src/components/overview/VariantsSummary.js:8]: neither need optional chaining im pretty sure, explorer should always at the very least be `{}`

**Comment by brouillette** [comment]
> Looks good, although gohan won't let me test right now. Also, when gohan fails, the spinner runs forever, not sure if this is a bug or a feature.

Patched the edge case for when gohan request fails

File: src/components/overview/VariantsSummary.js (hunk 1 above)
Comment (brouillette) [reply src/components/overview/VariantsSummary.js]: Patched the edge case for when gohan request fails

File: src/components/overview/VariantsSummary.js [hunk 2]
```diff
@@ -4,17 +4,24 @@ import { Col, Row, Spin, Statistic, Typography, Icon } from "antd";
 
 const VariantsSummary = ( ) => {
 
-    const fetchingTableSummaries = useSelector(state => state.tableSummaries?.isFetching);
-    const numVCFs =
-        useSelector((state) => state.overviewSummary.data?.data_type_specific?.experiment_results.file_format.VCF) || 0;
+    const fetchingVariantsOverview = useSelector(state => state.explorer.fetchingVariantsOverview);
+    const variantsOverviewResults = useSelector((state) => state.explorer.variantsOverviewResponse);
```
Comment (davidlougheed) [inline src/components/overview/VariantsSummary.js:8]: use `(state)` or `state` for single param but not both

**Review by davidlougheed — CHANGES_REQUESTED** [review]

**Review by brouillette — COMMENTED** [review]
patched

File: src/modules/explorer/reducers.js [hunk 3]
```diff
@@ -54,7 +55,18 @@ export const explorer = (
         case PERFORM_GET_GOHAN_VARIANTS_OVERVIEW.RECEIVE:
             return {
                 ...state,
-                variantsOverviewResponse: action.data
+                variantsOverviewResponse: action.data,
+            };
+        case PERFORM_GET_GOHAN_VARIANTS_OVERVIEW.REQUEST:
+            return {
+                ...state,
+                fetchingVariantsOverview: true
```
Comment (davidlougheed) [inline src/modules/explorer/reducers.js:63]: trailing comma

File: src/modules/explorer/reducers.js [hunk 4]
```diff
@@ -54,7 +55,18 @@ export const explorer = (
...
+            };
+        case PERFORM_GET_GOHAN_VARIANTS_OVERVIEW.FINISH:
+        case PERFORM_GET_GOHAN_VARIANTS_OVERVIEW.ERROR:
+            return {
+                ...state,
+                fetchingVariantsOverview: false
```
Comment (davidlougheed) [inline src/modules/explorer/reducers.js:69]: finish is called after error so you dont need the error line here

Comment (davidlougheed) [reply src/modules/explorer/reducers.js:69]: also add trailing comma

**Review by davidlougheed — CHANGES_REQUESTED** [review]
minor changes, otherwise looks good

**Review by davidlougheed — APPROVED** [review]
lgtm
